sql_file3.sql

```diff
@@ -0,0 +1,705 @@
+
+ INSERT INTO COMMENTS (Comments) VALUES
+('        This is a great product!'),
+('  I had an issue with the installation process.'),
+('Customer service was very helpful.'),
+('   Delivery was delayed by two days, but overall satisfied.'),
+('        Highly recommend to anyone looking for quality.'),
+('The user manual could be more detailed.'),
+('  Packaging was excellent, and the product arrived in good condition.'),
+('   The product exceeded my expectations. Will buy again.'),
+(' Had a minor issue, but it was resolved quickly.'),
+('     Great value for the price.');
+
+SELECT * FROM Comments;
+
+UPDATE Comments
+SET Comments= LTRIM(Comments);
+
+
+-- 33 Display the current date in the format 'dd-MM-yyyy'
+
+SELECT FORMAT(GETDATE(),'dd-MM-yyyy') AS CurrentDate;
+
+-- 34 From an 'Orders' table with an 'OrderTotal' column, display the total as a currency with 2 decimal places
+
+CREATE TABLE Orders2 (
+    OrderID INT PRIMARY KEY,
+    OrderTotal DECIMAL(10, 2)  -- 10 digits total, 2 decimal places
+);
+
+INSERT INTO Orders2 (OrderID, OrderTotal) VALUES
+(1, 1234.56),
+(2, 789.99),
+(3, 1500.00),
+(4, 99.95),
+(5, 2499.49);
+
+SELECT FORMAT(OrderTotal, 'C', 'en-US') AS FormattedTotal
+FROM Orders2;
+
+
+--35 Separate the string 'apple,banana,cherry' into individual fruits
+
+SELECT value AS Fruit
+FROM STRING_SPLIT('apple,banana,cherry', ',');
+
+-- 36 From a 'Skills' table with a 'SkillList' column containing comma-separated skills, write a query to create a row for each individual skill.
+CREATE TABLE Skills (
+    SkillID INT PRIMARY KEY,
+    SkillList VARCHAR(255)
+);
+
+INSERT INTO Skills (SkillID, SkillList) VALUES
+(1, 'SQL,Python,Data Analysis'),
+(2, 'Java,C++,JavaScript'),
+(3, 'HTML,CSS,React');
+
+
+SELECT SkillID, VALUE AS Skill
+FROM Skills
+CROSS APPLY STRING_SPLIT(SkillList, ',');
+
+
+-- Date and Time Functions
+
+
+-- 37 Write a query to display the current date and time.
+
+SELECT GETDATE() AS CurrentDateTime;
+
+-- 38 From an 'Orders' table, find all orders placed in the last 24 hours
+CREATE TABLE Orders3 (
+    OrderID INT PRIMARY KEY,
+    OrderTotal DECIMAL(10, 2),
+    OrderDate DATETIME  
+);
+
+INSERT INTO Orders3 (OrderID, OrderTotal, OrderDate) VALUES
+(1, 150.00, GETDATE() - 1),  
+(2, 75.50, GETDATE() - 0.5),  
+(3, 120.25, GETDATE() - 0.1),  
+(4, 250.00, GETDATE() - 0.02),  
+(5, 99.99, GETDATE());          
+
+
+SELECT *
+FROM Orders3
+WHERE OrderDate >= DATEADD(HOUR, -24, GETDATE());
+
+-- 39  Display the current UTC date and time
+SELECT GETUTCDATE() AS CurrentUTCTime;
+
+-- 40 Write a query to show the time difference between local time and UTC time
+
+SELECT SYSDATETIMEOFFSET() AS LocalDateTime, GETUTCDATE()  AS UtcDT , DATEDIFF(minute,SYSDATETIMEOFFSET(),GETUTCDATE()) AS DifferenceInMinutes;
+
+-- 41 Convert the current date and time to 'Pacific Standard Time' 
+
+SELECT SYSDATETIME() AS CurrentLocalTime,SYSDATETIMEOFFSET() AT TIME ZONE 'Pacific Standard Time' AS PacificTime;
```

Comment:
(https://learn.microsoft.com/en-us/sql/t-sql/queries/at-time-zone-transact-sql?view=sql-server-ver16#b-convert-values-between-different-time-zones)

First we need to specify the time zone of our stored data, then specify the target time zone.